week2/v2/index.js

```diff
@@ -0,0 +1,176 @@
+            break;    
+    }
+}
+
+function fileWriter(parsedData, fileName){
+    fs.writeFile(fileName, JSON.stringify(parsedData), (err) => {
+        if (err) throw err;
+    })
+}
+
+function manualReader(){
+    fs.readFile('./help.txt', 'UTF-8', (err, data) => {
+        if(err) throw err;
+        console.log(data);
+    });
+}
+
+function checkFileExistance(){
+    return new Promise((resolve, reject) => {
+        fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+            if (err) reject(err);
+            data = JSON.parse(data);
+            resolve(data);
+            return;
+        });
+    });
+
+}
+
+function addTodoItem(data, item){
+    if (item !== undefined){
+        data.push({'id': item, "status": ''})
+        fileWriter(data, './todos.json');
+    } else {
+        console.log('Specify what you want to add! Check "node index.js help" for more information');
+    }
+    
+}
+
+function itemsResetter(){
+    fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+        if (data !== '[]'){
+            data = JSON.parse(data);
+            data.length = 0;
+            fileWriter(data, './todos.json');
+            return;
+        }
+        console.log('already empty!');
+    });
+    
+}
+
+function listTodoItems(){
+    fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+        data = JSON.parse(data);
+        if(data !== '[]'){
+            let text = data.reduce(((prev, curr, index) => prev +=`${index + 1}. ${curr.id}\t\t\t${curr.status}\n`), ``);
+            console.log(text);
+            return;
+        }
+        console.log('Todo list is empty!');
+    });
+}
+
+function setMark(index, state){
+    index = parseInt(index) - 1;
+    fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+        data = JSON.parse(data);
+        if(data[index] !== undefined && state !== undefined){
+            data[index].status = state.toUpperCase();
+            fileWriter(data, './todos.json');
+        } else if(data[index] === undefined) {
+            console.log('No corresponding item!');            
+        } else {
+            console.log('Choose the status the you want, check "node index.js help" to see how to use it');            
+        }
+    }); 
+}
+
+function removeMark(index){
+    index = parseInt(index) - 1;
+    fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+        data = JSON.parse(data);
+        if(data[index] !== undefined && data[index].status !== '' ){
+            data[index].status = '';
+            fileWriter(data, './todos.json');
+        } else if(data[index] === undefined) {
+            console.log('No correspoding item!');            
+        } else{
+            console.log('There\'s no mark! Try "node index.js list" to see the items');            
+        }
+    });    
+}
+
+function updateItem(index){
+    index = parseInt(index) - 1;
+    fs.readFile('./todos.json', 'UTF-8', (err, data) => {
+        data = JSON.parse(data);
```

Comment:
Be careful about re-assigning variables. This works in javascript but not in all languages, and can be a bit of a bad practice anyway - it can lead to confusion about what's actually inside that variable. Better to use a new variable name for after it's been parsed.

Either `let data = JSON.parse(rawData)` or `let parsedData = JSON.parse(data)`, something like that.

Reply:
I agree, I was wondering myself but then I thought, less names should be better haha. I will modify it.